### ISSUE TITLE
TestCore no longer used. Slim down and integrate or remove?

### ISSUE THREAD

**Issue by mjftw** [Witekio/pluma-automation#89]
**Is your feature request related to a problem? Please describe.**
`TestCore` used to be included by default on all tests, but we've moved away from this architecture in favour of having all the individual steps be separately runnable.
I think there is definitely still some value in `TestCore`, but it likely needs to be slimmed down and integrated properly as part of the `TestSuite`.
`TestCore` can be found here: https://github.com/Witekio/pluma-automation/blob/master/pluma/test/test.py#L90

**Describe the solution you'd like**
I think there is definitely still some value in `TestCore`, but it likely needs to be slimmed down and integrated properly as part of the `TestSuite`.
`TestCore` can be found here: https://github.com/Witekio/pluma-automation/blob/master/pluma/test/test.py#L90

**Describe alternatives you've considered**
Alternatively it could be removed altogether if there really is no value here.

**Comment by Pixep** [comment]
I agree that this is hard to use in the current model. Some of it can be ported to TestBase, mainly standard setup and cleanup functions before/after the test we run. Otherwise the approach is different: single atomic tests, versus tests that can impact boot/flashing/etc., so not that easy to mix
